poppy/fresnel.py

```diff
@@ -636,14 +676,20 @@ def _propagate_ptp(self, dz):
             return
 
         x, y = self.coordinates()  # meters
-        rhosqr = np.fft.fftshift((x / (self.pixelscale.to(u.m / u.pix).value ** 2 * self.n)) ** 2 + (
-                                  y / (self.pixelscale.to(u.m / u.pix).value ** 2 * self.n)) ** 2)
-        t = -1.0j * np.pi * self.wavelength.to(u.meter).value * (
-            z_direct) * rhosqr  # Transfer Function of diffraction propagation eq. 22, eq. 87
+        meter_per_pix = self.pixelscale.to(u.m / u.pix).value
+        rhosqr = np.fft.fftshift((x / ( meter_per_pix** 2 * self.n)) ** 2 + (
```

Comment:
Should this use your `_fftshift` function here?

Reply:
yep.